0000-product.md

```diff
@@ -0,0 +1,463 @@
+
+``` 
+message ProductDeleteAction { 
+    enum Product_Namespace { 
+        UNSET_NAMESPACE = 0;
+        GS1 = 1; 
+    }
+    // product_namespace and product_id are used in deriving the state address
+    Product_Namespace product_namespace = 1; 
+    string product_id = 2; 
+} 
+```
+
+If the grid setting grid.product.allow_delete is set to false, this transaction
+is invalid. The default value for grid.product.allow_delete is true. This
+setting is stored using the Sawtooth Settings smart contract, more information
+can be found
+[here](https://sawtooth.hyperledger.org/docs/core/releases/latest/transaction_family_specifications/settings_transaction_family.html).
+
+Validation requirements:
+
+* If a product with product_id does not exist the transaction is invalid.  
+* The signer of the transaction must be an agent in the Pike state and must
+belong to an organization in Pike state, otherwise the transaction is invalid.
+* The owner in the product must match the organization that the agent belongs to, otherwise the transaction is invalid.  
+* The agent must have the permission “can_delete_product” for the organization,
+otherwise the transaction is invalid.
+
+The inputs for ProductDeleteAction must include:
+
+* Address of the Agent submitting the transaction Address of the Organization
+* the Product is being deleted for Address of the Product to be deleted
+
+The outputs for ProductUpdateAction must include:
+
+* Address of the Product to be deleted
+
+### Defined GS1 Properties
+
+An initial set of GS1 properties will be predefined within Grid. Each property
+will have a property definition of the following format:
+
+``` 
+PropertyDefinition( 
+    name="<GS1 Property>",
+    data_type=PropertyDefinition.DataType.STRING, 
+    required=False, 
+    description="A description of the GS1 data.") 
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Each GTIN format, except GTIN-8, contains an GS1 Company Prefix organizational
+identifier encoded into it, and this RFC duplicates the organization
+identification with a Grid-specific owner field to tie the product to
+organizations managed in Pike. In the future, it may be good to use the GTIN’s
+organizational identifier directly.
+
+Currently, Pike does not provide a place to store a GS1 Company Prefix, and the
+permissions around who is authorized to provide this prefix is complicated and
+not a direct extension of how Pike currently works. As such it is out of the
+scope of this RFC.
+
+Instead, Pike’s organization definition will be extended to include metadata,
+similar to Agent’s metadata. Metadata is a Key,Value store of arbitrary data.
+The GS1 Company Prefix will be stored in the metadata under
+“gs1_company_prefix”.
+
+Solving how to properly provide GS1 Company Prefixes to a Pike Organization will
+be solved in a future RFC.
+
+Trade items can include non-material goods, such as services, which will also
+need to be represented within Grid. This is not covered by this RFC.
+
```

Comment:
I think we should be more specific about this. 
1. Item/Product - "Class level identification = GTIN" IN Scope "Batch/Lot (LGTIN) or Serialized (SGTIN) are OUT of scope. We should discuss what this means and if there are any other type of data that are required to be associated with a specific instance of a product (temp data, expiry date..etc)
2. Logistical Cases/Pallets  -  Higher levels of packaging of the items - some of which are the trade unit between to parties although not necessarily the consumer unit. Identified sometimes with a GTIN but often times, due to their heterogeneous nature, with another GS1 identifier (SSCC).  
3. Assets (returnable or fixed)  - Identified with GRAI or GIAI - I presume like services (GSRN)  OUT

Reply:
I've added more specific descriptions about non-material goods base on your feedback. 

1. LGTIN and SGTIN are out of scope. Grid product is extensible to include additional fields like temp data, expiry date, etc (configurable on a use case basis)

2. Great information to know for a future iteration of Grid Product

3. Also good to know, can you give some examples of what is meant by "Assets", so I can include that feed back in the RFC

Reply:
My perspective is SGTIN/LGTIN are different Schemas and shoudl be different RFC.